estimator/lwe_primal.py

```diff
@@ -300,10 +300,20 @@ def gaussian_heuristic_log_input(r):
 
         d = len(r)
         r = [log(x) for x in r]
-        for i, _ in enumerate(r):
-            if gaussian_heuristic_log_input(r[i:]) < D.stddev**2 * (d - i):
-                return ZZ(d - (i - 1))
-        return ZZ(2)
+
+        if d > 4096:
+            for i, _ in enumerate(r):
+                # chosen arbitrarily
+                j = d - 1024 + i
```

Comment:
So this is saying: start with block size 1024 max?

Reply:
For the final SVP call in size `eta` yes, so instead of looking for `eta \in [n, n-1, ..., 2]` it would be `eta \in [1024, 1023,...2]` -- I admit it's a bit unsatisfactory. 

Maybe I can look at what the required `eta` would be for e.g `RC.ADPS16(eta, 4096) > 2**512` or something, and use this as an upper bound instead?

Reply:
will update!